source/dub/generators/build.d

```diff
@@ -529,19 +529,41 @@ class BuildGenerator : ProjectGenerator {
 				if (!exe_path_string.startsWith(".") && (exe_path_string.length < 2 || exe_path_string[1] != ':'))
 					exe_path_string = ".\\" ~ exe_path_string;
 			}
+			runPreRunCommands(m_project.rootPackage, m_project, settings, buildsettings);
 			logInfo("Running %s %s", exe_path_string, run_args.join(" "));
 			if (settings.runCallback) {
 				auto res = execute(exe_path_string ~ run_args);
 				settings.runCallback(res.status, res.output);
+				runPostRunCommands(m_project.rootPackage, m_project, settings, buildsettings);
```

Comment:
shouldn't targetExitStatus also be set here before the run callback?

Reply:
you are right, fixed